### DISCUSSION TITLE
How to handle join/split behaviour for attributes ?

### DISCUSSION THREAD

**Discussion by Granipouss** [ueberdosis/tiptap#4578] [General Questions]
Hey everyone,

I have been using Tiptap and it's been great, but I need some help on the attribute management.

I declared custom numeric attribute added the a block Node and I want to have control on the way it is generate when splitting or joining nodes. It seems that the behaviour is currently to  copy attributes on split and using only the first one on join. Instead I would like it to be this attribute to also be split in half on split (10 becomes 5 and 5) and added on join (4 and 3 become 7) for instance.

Is there a simple way to do such thing ?

So far I tried to watch transactions but it wasn't very successful and seems very hacky. My best attempt is to override the keymap for Enter and Backspace and basically rewrite the associated command splitBlock and joinBackward, but it seems really overkill, given it seems like there is no way to extend a command. I really feel like I am missing something.

If that is more of ProseMirror issue, I'll create an issue there.

Cheers,

**Comment by semiaddict** [comment]
Hi @Granipouss,
I know this is an old issue, but I have a very similar reqiurement. Did you even manage to find a solution ?
Thank you in advance.
